### ISSUE TITLE
Fatal error accessing modules that contain macros

### ISSUE THREAD

**Issue by yallop** [modular-macros/ocaml-macros#13]
```
$ cat mod.ml 
macro f x = << $x + 1 >>
$ ocamlc -c mod.ml
$ ocaml
        OCaml version 4.04.0+dev9-2015-09-05

# macro g = ~Mod.f;;
Fatal error: exception File "typing/env.ml", line 1515, characters 11-17: Assertion failed
$ 
```

**Comment by OlivierNicole** [comment]
I think this should not typecheck: `~Mod.f` is of phase 2, whereas macros are of phase 1. (Using the definition of phase of the implementation, the one that we used back in 2017, I don't know how it is formalised nowadays.)

It should not fail with a fatal error, though.

**Comment by yallop** [comment]
This is fixed as of c1d7fc74b7c55515bf9435ae07ffee89384f6689:

```
# macro g = ~Mod.f;;
Characters 10-16:
  macro g = ~Mod.f;;
            ^^^^^^
Error: Attempt to use value ~Mod.f of phase 2 in an environment of phase 1
```
